Game of Life (#9)
anaplan-engineering/kazuki: gameoflife/README.md

# Game of Life

An example specification of Conway's Game of Life in Kazuki, based on a version written

> in [VDM](https://www.overturetool.org/download/examples/VDMSL/ConwayGameLifeSL/).

Reviewer: 🚫 I would prefer better attribution:
```suggestion
An example specification of Conway's Game of Life in Kazuki, based on a [VDM version](https://github.com/overturetool/overturetool.github.io/blob/master/download/examples/VDMSL/ConwayGameLifeSL/index.md) by Nick Battle, Peter Gorm Larsen and Claus Ballegaard Nielsen.
```
Note that in the suggestion above I've replaced the `www.overturetool.org` URL with the GitHub URL for the source of the website, which is going to be more stable.

Conway's Game of Life is a theoretical system embedded into a two-dimensional square grid.
Each square is either alive or dead (sometimes called "empty").
Each new generation is determined based only on the current generation's state.

> Each square on the grid interacts with it's eight neighbours (horizontal, vertical and diagonal).

Reviewer: 🚫
```suggestion
Each square on the grid interacts with its eight neighbours (horizontal, vertical and diagonal).
```

The rules for this are as follows:

- Any live square with less than two live neighbours dies
- Any live square with more than three live neighbours dies

> - Any live square with two or three neighbours survives into the next generation

Reviewer: ⛏️ I don't really mind whether you say "live neighbours" or just "neighbours", but the phrasing should be consistent across all four rules.
```suggestion
- Any live square with two or three live neighbours survives into the next generation
```

> - Any empty square with three neighbours will become a live square

Reviewer: ⛏️ 
```suggestion
- Any empty square with three live neighbours will become a live square
```

> The system is given an intial state uses that to calculate the subsequent generations.

Reviewer: 🚫 Sentence needs rephrasing, or at least a connective between "state" and "uses".
